avro/src/test/scala/zio/blocks/avro/AvroFormatSpec.scala

```diff
@@ -0,0 +1,64 @@
+package zio.blocks.avro
+
+import zio.Scope
+import zio.blocks.schema.Schema
+import zio.test.Assertion._
+import zio.test._
+import java.nio.ByteBuffer
+import java.util
+import java.util.UUID
+
+object AvroFormatSpec extends ZIOSpecDefault {
+  def spec: Spec[TestEnvironment with Scope, Any] = suite("AvroFormatSpec")(
+    suite("primitives")(
+      test("String") {
+        roundTrip("Hello", 6)
+      },
+      test("Boolean") {
+        roundTrip(true, 1)
+      },
+      test("Int") {
+        roundTrip(42, 1)
+      },
+      test("Short") {
+        roundTrip(42: Short, 1)
+      },
+      test("Long") {
+        roundTrip(600000000000L, 6)
+      },
+      test("Float") {
+        roundTrip(42.0f, 4)
+      },
+      test("Double") {
+        roundTrip(42.0, 8)
+      },
+      test("Char") {
+        roundTrip('a', 2)
+      },
+      test("UUID") {
+        roundTrip(UUID.randomUUID(), 37)
+      },
+      test("Currency") {
+        roundTrip(java.util.Currency.getInstance("USD"), 4)
+      }
+    )
+  )
+
+  def roundTrip[A: Schema](value: A, expectedLength: Int): TestResult = {
+    val result = encodeToByteArray(out => Schema[A].encode(AvroFormat)(out)(value))
+    assert(result.length)(equalTo(expectedLength)) &&
+    assert(Schema[A].decode(AvroFormat)(toHeapByteBuffer(result)))(isRight(equalTo(value))) &&
+    assert(Schema[A].decode(AvroFormat)(toDirectByteBuffer(result)))(isRight(equalTo(value)))
+  }
+
+  def encodeToByteArray(f: ByteBuffer => Unit): Array[Byte] = {
+    val byteBuffer = ByteBuffer.allocate(1024)
```

Comment:
`ByteBuffer` needs to be initialized with the right size which we do not know when we call `encode`, wouldn't it be better to change `encode` to `def encode(value: Value): EncodeOutput`? Or switch to `ByteArrayOutputStream`?

Otherwise we are forced to use an arbitrary large size.